### DISCUSSION TITLE
How to prevent archiveUrl from appearing in yarn.lock?

### DISCUSSION THREAD

**Discussion by grubeninspekteur** [yarnpkg/berry#3012] [Q&A]
Hi,

we use different registries locally and at the client side. When we run yarn install locally (against Nexus), an __archiveUrl is added to the resolution keys. This results in  `getaddrinfo ENOTFOUND` on the CI site that does not have acess to the specified registry.

I tried removing the __archiveUrl as a prepare step of the pipeline, however then `yarn install --immutable` may fail as it wants to merge entries:

```
52098 [INFO] ➤ YN0028: │ -"depd@npm:^1.1.2":
52098 [INFO] ➤ YN0028: │ +"depd@npm:^1.1.2, depd@npm:~1.1.2":
52098 [INFO] ➤ YN0000: │    version: 1.1.2
52099 [INFO] ➤ YN0000: │    resolution: "depd@npm:1.1.2"
52099 [INFO] ➤ YN0000: │    checksum: f45566ff7019a346852f095768a380778ed544de24e103b479fd5d3e61982d670efbb5234c09d0588d7fdb09c26c48283d7150e4be5e6ce5d3d37cd268d75c4d
52099 [INFO] ➤ YN0000: │    languageName: node
52099 [INFO] ➤ YN0000: │    linkType: hard
52099 [INFO] ➤ YN0000: │ 
52099 [INFO] ➤ YN0028: │ -"depd@npm:~1.1.2":
52099 [INFO] ➤ YN0028: │ -  version: 1.1.2
52099 [INFO] ➤ YN0028: │ -  resolution: "depd@npm:1.1.2"
52099 [INFO] ➤ YN0028: │ -  checksum: f45566ff7019a346852f095768a380778ed544de24e103b479fd5d3e61982d670efbb5234c09d0588d7fdb09c26c48283d7150e4be5e6ce5d3d37cd268d75c4d
52102 [INFO] ➤ YN0028: │ -  languageName: node
52102 [INFO] ➤ YN0028: │ -  linkType: hard
```

Though this could be an artifact from the migration. After removing the URLs locally and running `yarn install`, the entries were merged as well.

Still - is there a way to tell Yarn not to add the URL at all? Is the archiveUrl really necessary?

**Comment by arcanis** [comment]
The `__archiveUrl` annotation is caused by your registry having a non-standard format for tarball urls. Yarn expects the following:

```
<registry endpoint>/<@scope/name>/-/<name>-<version>.tgz
```

If the registry uses something else than this format, then we cannot just store the version; we need the full url.

**Reply by grubeninspekteur** [reply]
But shouldn't lockfile information be based solely on the version of the package (+checksum)? Yarn is able to resolve the package just fine with --immutable when I remove the archiveUrl, sans the above semantic issues how the lockfile is structured. (Please forgive my ignorance of the actual resolution process)

We are evaluating the migration to Yarn 2 because of issue [#5892](https://github.com/yarnpkg/yarn/issues/5892) , which is marked as fixed for Berry.
